api/api.go

```diff
@@ -201,16 +201,21 @@ func (a *API) GetPlugins(context.Context, *emptypb.Empty) (*v1.PluginConfigs, er
 }
 
 // GetPools returns the pool configuration of the GatewayD.
-func (a *API) GetPools(context.Context, *emptypb.Empty) (*structpb.Struct, error) {
-	_, span := otel.Tracer(config.TracerName).Start(a.ctx, "Get Pools")
+func (a *API) GetPools(ctx context.Context, _ *emptypb.Empty) (*structpb.Struct, error) {
+	_, span := otel.Tracer(config.TracerName).Start(ctx, "Get Pools")
```

Comment:
Do not use request context for tracing here. If you do, then change all instances and remove the `a.ctx` everywhere. To be fair, the whole tracing should be refactored to have different traces for different clients and plugins.

```suggestion
func (a *API) GetPools(context.Context, _ *emptypb.Empty) (*structpb.Struct, error) {
	_, span := otel.Tracer(config.TracerName).Start(a.ctx, "Get Pools")
```